certbot/plugins/common.py

```diff
@@ -99,7 +100,131 @@ def dest(self, var):
     def conf(self, var):
         """Find a configuration value for variable ``var``."""
         return getattr(self.config, self.dest(var))
-# other
+
+
+class PluginStorage(object):
+    """Class implementing storage functionality for plugins"""
+
+    def __init__(self, config, classkey):
+        """Initializes PluginStorage object storing required configuration
+        options.
+
+        :param .configuration.NamespaceConfig config: Configuration object
+        :param str classkey: class name to use as root key in storage file
+
+        :returns: Plugin storage object
+        :rtype: dict
+
+        :raises .errors.PluginStorageError: when unable to open or read the file
+        """
+
+        self.config = config
+        self.classkey = classkey
+        self.initialized = False
+        self._data = None
+        self.storagepath = None
+
+    def initialize_storage(self):
```

Comment:
Is there ever a time we call this function and `os.path.isdir(self.config.config_dir)` not being true isn't an error? Looking at where it's called, we always immediately try to use `self._data` which would otherwise be `None` and raise an error. Perhaps we should remove the `else` branch here.

Similar to Erica's comments, I'm not sure if a user of this class would/should ever call this function, in which case it should be private.